runtime/lua/vim/lsp.lua

```diff
@@ -890,6 +890,42 @@ function lsp.start_client(config)
     end)
   end
 
+  --@private
+  --- Sends a request to the server and synchronously waits for the response.
+  ---
+  --- This is a wrapper around {client.request}
+  ---
+  --@param method (string) LSP method name.
+  --@param params (table) LSP request params.
+  --@param timeout_ms (number, optional, default=100) Maximum time in
+  ---milliseconds to wait for a result.
+  --@param bufnr (number) Buffer handle (0 for current).
+  --@returns { err, result }, where `err` and `result` come from the |lsp-handler|.
```

Comment:
```suggestion
  --@returns { err, result }, where `err` and `result` come from the |lsp-handler|.
```

By reading the docs I thought this would be a list where the first item is the error and the second item is the result. But reading the code I figured it is a dictionary with `err` and `result` as keys. Maybe that could be clarified in the docs?

Reply:
Indeed. It's the same format as the `request_result` from `buf_request_all` and `buf_request_sync`, where the structure is not documented at all.

Something like `{ ["error"] = error, ["result"] = result }` or `{ error = error, result = result }` is verbose but should make things clear. What do you think?